connectors/sources/azure_blob_storage.py

```diff
@@ -191,16 +200,34 @@ async def get_content(self, blob, timestamp=None, doit=None):
         ) as blob_client:
             data = await blob_client.download_blob()
             temp_filename = ""
-            async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
-                temp_filename = async_buffer.name
-                async for content in data.chunks():
-                    await async_buffer.write(content)
-            self._logger.debug(f"Calling convert_to_b64 for file : {blob_name}")
-            await asyncio.to_thread(convert_to_b64, source=temp_filename)
-            async with aiofiles.open(file=temp_filename, mode="r") as async_buffer:
-                # base64 on macOS will add a EOL, so we strip() here
-                document["_attachment"] = (await async_buffer.read()).strip()
-            await remove(str(temp_filename))
+
+            try:
+                async with NamedTemporaryFile(
+                    mode="wb",
+                    suffix=file_extension,
+                    delete=False,
+                    dir=self.download_dir,
+                ) as async_buffer:
+                    temp_filename = async_buffer.name
+                    async for content in data.chunks():
+                        await async_buffer.write(content)
+
+                if self.configuration["use_text_extraction_service"]:
+                    if self.extraction_service._check_configured():
+                        document["body"] = await self.extraction_service.extract_text(
+                            temp_filename, blob_name
+                        )
+                else:
+                    self._logger.debug(f"Calling convert_to_b64 for file : {blob_name}")
+                    await asyncio.to_thread(convert_to_b64, source=temp_filename)
+                    async with aiofiles.open(
+                        file=temp_filename, mode="r"
+                    ) as async_buffer:
+                        # base64 on macOS will add a EOL, so we strip() here
+                        document["_attachment"] = (await async_buffer.read()).strip()
+            finally:
+                await remove(str(temp_filename))
```

Comment:
Is this possible to move up, so that each source does not have to do this?

Can be done later, probably together with making downloads in framework more generic

Reply:
Good idea, I'll move it up